rskj-core/src/main/java/co/rsk/cli/tools/ImportState.java

```diff
@@ -35,20 +37,32 @@
 /**
  * The entry point for import state CLI tool
  * This is an experimental/unsupported tool
- *
+ * <p>
  * Required cli args:
  * - args[0] - file path
  */
 public class ImportState extends CliToolRskContextAware {
 
+    private static final Logger logger = LoggerFactory.getLogger("ImportState");
+
     public static void main(String[] args) {
         create(MethodHandles.lookup().lookupClass()).execute(args);
     }
 
     @Override
     protected void onExecute(@Nonnull String[] args, @Nonnull RskContext ctx) throws Exception {
-        String databaseDir = ctx.getRskSystemProperties().databaseDir();
-        KeyValueDataSource trieDB = LevelDbDataSource.makeDataSource(Paths.get(databaseDir, "unitrie"));
+        RskSystemProperties rskSystemProperties = ctx.getRskSystemProperties();
+        String databaseDir = rskSystemProperties.databaseDir();
+        KeyValueDataSource trieDB = KeyValueDataSource.makeDataSource(Paths.get(databaseDir, "unitrie"), rskSystemProperties.databaseKind());
+
+        try {
+            KeyValueDataSource.validateDbKind(rskSystemProperties.databaseKind(), rskSystemProperties.databaseDir(), rskSystemProperties.databaseReset() || rskSystemProperties.importEnabled());
```

Comment:
we probably have to validate db kind before `KeyValueDataSource.makeDataSource()`? Otherwise, `KeyValueDataSource.makeDataSource()` will crash anyway.

Reply:
Done